### ISSUE TITLE
 Accuracy of this method:python main_rgb_fused.py recognition -c config/ntu60_xsub/test_rgb_fused.yaml

### ISSUE THREAD

**Issue by katahiyu** [bruceyo/MMNet#5]
Dear author,
I ran this code using a trained model and got the following results: Why is the Top1 accuracy 4.83%? Please let me know if you have any solution.
![image](https://user-images.githubusercontent.com/114911420/194708739-b4785eee-75f0-4a20-8bb6-8090f2a91c87.png)

**Comment by bruceyo** [comment]
The current trained models are from the conference version based on the ST-GCN's skeleton data (dating back to May 2020).
The journal version's implementations (around Jan 2021) are based on the MS-G3D's skeleton data, where the order is inconsistent with the one of ST-GCN.
Sorry about it. I will upload the newly retrained model. The trained models for other datasets should be correct.

You may try to retrain the model to get the proper results.

**Comment by katahiyu** [comment]
Thank you for your reply.
I understand about trained models. We would appreciate it if you could upload your retrained model.

I'm having trouble retraining the model.
 When I run
 "python main_rgb_fused.py recognition -c config/<dataset>/train_rgb_fused.yaml"
 it returns low accuracy like the image below.
Is there any solution for this?
![image](https://user-images.githubusercontent.com/114911420/194980629-94e7ecb9-315a-4501-8c87-2a63585d2a02.png)

I really appreciate your kind response.

**Comment by bruceyo** [comment]
Retained models `rgb_fused_model_new.pt` are uploaded to Google Drive. 
[NTU RGB+D X-Sub](https://drive.google.com/drive/folders/1H_8r51CIvM5kJwIxA9ThPuj4TzBDfGjA)
[NTU RGB+D X-View](https://drive.google.com/drive/folders/1oluZhCcGxGnrCtjOC-QbtfW4Hb2UcvRL)

**Comment by katahiyu** [comment]
Thanks for the upload.
Why does a new trained model result in lower accuracy?
![image](https://user-images.githubusercontent.com/114911420/195996523-6bd8f449-c619-40e2-a9a6-1fd1a1b2051e.png)

**Comment by bruceyo** [comment]
![20221017102013](https://user-images.githubusercontent.com/6397416/196075351-b6a73f7e-90b3-43c1-adce-3e0622cc823d.png)
Above is my results. 
I do not know your exact configuration. Probably due to the improper preparation of skeleton data.

**Comment by katahiyu** [comment]
Is the best_model.pt the same as the rgb_fused_model_new.pt you recently uploaded?

**Comment by bruceyo** [comment]
> Is the best_model.pt the same as the rgb_fused_model_new.pt you recently uploaded?

Yes, I downloaded and tested it again. I used the same skeleton data preparation with the MS-G3D.

**Comment by katahiyu** [comment]
Thanks for the reply. Understood.

This is my feeder_rgb_fused.py. is this path the correct place to put the processed ROI data downloaded from GoogleDrive? This may be a rudimentary question, but thank you in advance.
![image](https://user-images.githubusercontent.com/114911420/196107149-28440d69-90a6-40c9-8009-5de7dbee118e.png)

**Comment by bruceyo** [comment]
> Thanks for the reply. Understood.
> 
> This is my feeder_rgb_fused.py. is this path the correct place to put the processed ROI data downloaded from GoogleDrive? This may be a rudimentary question, but thank you in advance. ![image](https://user-images.githubusercontent.com/114911420/196107149-28440d69-90a6-40c9-8009-5de7dbee118e.png)

Yes, you can use it to reproduce the result. But not recommended for training as it is implemented with random sampling (see Line 131 of [feeder_rgb_fused_ntu.py).](https://github.com/bruceyo/MMNet/blob/main/feeder/feeder_rgb_fused_ntu.py)

**Comment by katahiyu** [comment]
I see!

It is still not accurate and does not run well, is there any part of the provided program code that needs to be changed?

**Comment by katahiyu** [comment]
> > 返信いただきありがとうございます。了解した。
> > これは私の feeder_rgb_fused.py です。このパスは、GoogleDrive からダウンロードした処理済みの ROI データを配置する正しい場所ですか? 初歩的な質問かもしれませんがよろしくお願いします。![画像](https://user-images.githubusercontent.com/114911420/196107149-28440d69-90a6-40c9-8009-5de7dbee118e.png)
> 
> はい、結果を再現するために使用できます。ただし、ランダム サンプリングで実装されているため、トレーニングにはお勧めしません ( [feeder_rgb_fused_ntu.py の 131 行目を参照)。](https://github.com/bruceyo/MMNet/blob/main/feeder/feeder_rgb_fused_ntu.py)

This is my feeder_rgb_fused.py file. If I comment out the path to the processed ROI data, I get the same result. Is this evidence that I am not importing the ROI data properly?
![image](https://user-images.githubusercontent.com/114911420/196365305-748753b4-d20c-4d29-9058-d55789234625.png)

**Comment by bruceyo** [comment]
> 

Can you try to change Line 131 of [feeder_rgb_fused_ntu.py](https://github.com/bruceyo/MMNet/blob/main/feeder/feeder_rgb_fused_ntu.py) to `if not self.evaluation:`

**Comment by katahiyu** [comment]
Thanks for the reply.

I changed line 131 in the feeder_rgb_fused_ntu.py file to "if not self.evaluation:" and got the following result. Thank you! 
![image](https://user-images.githubusercontent.com/114911420/196387057-76fd2c6b-4743-4776-be08-21e1d3c268eb.png)

**Comment by bruceyo** [comment]
> 

Great. The slight result difference might be caused by the calculation of the ST-ROI.
The provided ST-ROI is from the conference version.

**Comment by katahiyu** [comment]
Thank you so much for your help. Thank you for the instant and solution you provided.
I would be happy to learn more about your training.

**Comment by 2233950316** [comment]
@katahiyu Hello, I have been reproducing this mmnet recently and I have some basic questions that I would like to ask you. Can I provide you with a contact information? My vx is meng2233950316. Thank you very much

**Comment by VinnyCSY** [comment]
@bruceyo 
I want to train rgb_fused model for MS-G3D backbone. 
Can you show me the code for ms-g3d backbone? 
Or could you tell me how to change ST-GCN skeleton data to MS-G3D skeleton data in training process?
